src/handler/page.rs

```diff
@@ -74,14 +76,23 @@ impl PageInner {
         execute(cmd, self.sender.clone(), Some(self.session_id.clone())).await
     }
 
-    /// This responds with the final http response when the page is loaded
-    pub(crate) async fn wait_for_navigation(&self) -> Result<Option<Arc<HttpRequest>>> {
-        let (tx, rx) = oneshot_channel();
-        self.sender
-            .clone()
-            .send(TargetMessage::WaitForNavigation(tx))
-            .await?;
-        Ok(rx.await?)
+    /// Create a PDL command future
+    pub(crate) fn execute_future<T: Command>(&self, cmd: T) -> Result<CommandFuture<T>> {
+        CommandFuture::new(cmd, self.sender.clone(), Some(self.session_id.clone()))
+    }
+
+    /// This creates navigation future with the final http response when the page is loaded
+    pub(crate) fn navigation_future(&self) -> NavigationFuture {
+        NavigationFuture::new(self.sender.clone())
+    }
```

Comment:
renaming this is a breaking change of a commonly used function.

I'd prefer to keep `wait_for_navigation` because I believe its name is more concise here

Reply:
Makes sense the name is puzzling indeed